src/test/java/com/microsoft/graph/authentication/TokenCredentialAuthProviderTest.java

```diff
@@ -0,0 +1,111 @@
+package com.microsoft.graph.authentication;
+
+import com.azure.core.credential.TokenCredential;
+import com.microsoft.graph.exceptions.AuthenticationException;
+import com.microsoft.graph.http.IHttpRequest;
+import com.microsoft.graph.mocks.MockTokenCredential;
+import com.microsoft.graph.options.HeaderOption;
+
+import okhttp3.Request;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TokenCredentialAuthProviderTest {
```

Comment:
consider adding test cases for:
- valid host names
- malformed URLs

Reply:
and valid scheme

Reply:
- added scheme validation
- added tests for other national clouds
- moved to a prototype using URL instead of string for the validation method. That's because the information is coming from APIs that are already doing the parsing anyway. No point in parsing, serializing and parsing the URL again.
Thanks for all the suggestions!